src/python/grpcio_tests/tests_aio/unit/server_test.py

```diff
@@ -98,89 +174,167 @@ def service(self, handler_details):
 class TestServer(AioTestBase):
 
     async def setUp(self):
-        self._server_target, self._server, self._generic_handler = await _start_test_server(
-        )
+        addr, self._server, self._generic_handler = await _start_test_server()
+        self._channel = aio.insecure_channel(addr)
 
     async def tearDown(self):
+        await self._channel.close()
         await self._server.stop(None)
 
     async def test_unary_unary(self):
-        async with aio.insecure_channel(self._server_target) as channel:
-            unary_unary_call = channel.unary_unary(_SIMPLE_UNARY_UNARY)
-            response = await unary_unary_call(_REQUEST)
-            self.assertEqual(response, _RESPONSE)
+        unary_unary_call = self._channel.unary_unary(_SIMPLE_UNARY_UNARY)
+        response = await unary_unary_call(_REQUEST)
+        self.assertEqual(response, _RESPONSE)
 
     async def test_unary_stream_async_generator(self):
-        async with aio.insecure_channel(self._server_target) as channel:
-            unary_stream_call = channel.unary_stream(_UNARY_STREAM_ASYNC_GEN)
-            call = unary_stream_call(_REQUEST)
+        unary_stream_call = self._channel.unary_stream(_UNARY_STREAM_ASYNC_GEN)
+        call = unary_stream_call(_REQUEST)
 
-            # Expecting the request message to reach server before retriving
-            # any responses.
-            await asyncio.wait_for(self._generic_handler.wait_for_call(),
-                                   test_constants.SHORT_TIMEOUT)
+        # Expecting the request message to reach server before retriving
+        # any responses.
+        await asyncio.wait_for(self._generic_handler.wait_for_call(),
+                               test_constants.SHORT_TIMEOUT)
 
-            response_cnt = 0
-            async for response in call:
-                response_cnt += 1
-                self.assertEqual(_RESPONSE, response)
+        response_cnt = 0
+        async for response in call:
+            response_cnt += 1
+            self.assertEqual(_RESPONSE, response)
```

Comment:
Today I learn.

For this scenario, we need to check if the response is valid. Both of them are valid to me, the first one seems a little hacky due to using `==` instead of `assertEqual`. But they are essentially the same.

```Python
        valid_responses = [response async for response in call if _RESPONSE == response]
        self.assertEqual(_NUM_STREAM_RESPONSES, len(valid_responses))
```

```Python
        response_cnt = 0
        async for response in call:
            response_cnt += 1
            self.assertEqual(_RESPONSE, response)

        self.assertEqual(_NUM_STREAM_RESPONSES, response_cnt)
```

Reply:
I think you can simply do
```python
responses = [resp async for resp in call]
self.assertSequenceEqual([_RESPONSE] * _NUM_STREAM_RESPONSES, reponses)
```

But again, this is purely stylistic and therefore optional.